rest/rest_error.go

```diff
@@ -105,3 +379,12 @@ func parseErrors(prefix string, err map[string]any) string {
 
 	return strings.Join(s, "\n")
 }
+
+// IsRestErrorCode returns true if the error is a *Error with one of the given error codes
+func IsRestErrorCode(err error, codes ...ErrorCode) bool {
```

Comment:
I frequently find myself having to write this logic, so I think it would be nice to have this helper function directly in Disgo!